server/app/constants.py

```diff
@@ -0,0 +1,13 @@
+import enum
+
+
+class Limit(int, enum.Enum):
+    SMALL = 2**6  # 64
+    MEDIUM = 2**8  # 256
+    LARGE = 2**10  # 1024
+    MAXINT4 = 2**31  # Maximum value signed 32-bit integer + 1
+
+
+class Pattern(str, enum.Enum):
+    SENSOR_IDENTIFIER = r"^(?!-)(?!.*--)[a-z0-9-]{1,64}(?<!-)$"
```

Comment:
It looks complicated, but it isn't really. It's the same as `^[a-z0-9-]{1,64}$`, with the extra constraint that there can't be `-` in the first or last position, and no consecutive `-`. These are the often used naming constraints in popular services.

The reason why I want to constrain the `-` is that databases also often constrain these in the same way. In case I want to name a column e.g. after the `value_identifier`.

If you need an easy way to explain this to someone: "lowercase letters and numbers, separated by dashes".

Reply:
OK, when you are using these for column names, then this definitely makes sense because the restriction actually has a purpose.

I just didn't want to have it like these unnecessary password requirements (one special character, a number, your first newborn, etc.).